packages/app/src/cli/services/dev/app-events/file-watcher.ts

```diff
@@ -221,8 +221,11 @@ export class FileWatcher {
           this.pushEvent({type: 'app_config_deleted', path, extensionPath, startTime})
         } else if (isExtensionToml) {
           // When a toml is deleted, we can consider every extension in that folder was deleted.
-          this.extensionPaths = this.extensionPaths.filter((extPath) => extPath !== extensionPath)
-          this.pushEvent({type: 'extension_folder_deleted', path: extensionPath, extensionPath, startTime})
+          // We need to wait in case this is actually just moving folders around, not deleting them.
```

Comment:
What happens if we don't wait? If it appears as removed and then as added, I don't think it's a big issue and also I guess it won't be very common. So maybe it's better to keep the code simpler.